submissions/kasionio/oop-exercise/index.js

```diff
@@ -0,0 +1,101 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/kasionio/a-tiny-JS-world/tree/gh-pages
+   Web app: https://kasionio.github.io/a-tiny-JS-world
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+  constructor(species, name, gender, legs, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying;
+      this.friends = [];
+  }
+
+  addFriends = (...friendsInstances) => this.friends.includes(...friendsInstances) ? false : this.friends = [...this.friends, ...friendsInstances];
+
+  getFriendsNames = () => this.friends.length > 0 ? this.friends.map(friend => friend.name) : 'forever alone...';
+
+  toString() {
```

Comment:
1. Mixing style of methods definition is not really a good style. Please fix.

2. `addFriends` method implementation:
```
this.friends.includes(...friendsInstances) ? false : this.friends = [...this.friends, ...friendsInstances];
```
Really hectic fragment.
The method returns false if all new friends are already in the friend list. It will add friends if they are all new and also return an updated friends list. So it returns boolean under condition and an array (potentially empty) in another case. This inconsistency will make the receiving party to add extra business logic to determine if the operation was successful.

We'd rather want just add any new friends.

Consider the following example:
`list = [...list, ...(newEntries.filter(newEntry => list.includes(entry)))]`
Please comprehend it before embedding. Feel free asking questions.

3. Ternary
Any assignments within the ternary like `condition ? variable = value1 : variable = value2`(and variations thereof)  is an anti-pattern. The pattern is to assign the entire ternary result to a variable (or return, or use otherwise).
Please check the proper use of ternary operator on MDN.

Reply:
Not necessarily assign before return. Assignment can be redundant of the value returned immediately after.
The point was not to have any assignments **inside** the ternary. `... ? .... : var = value` is a an anti-pattern.

Reply:
Thank you! Ternary is a bit confusing to me. I understand that we need assign the whole ternary to variable or have some functions inside ternary. At this case if condition `this.friends.length > 0` return false can we just return string value `alone` without assigning? Is it valid?
 `return (this.friends.length > 0 ? this.friends.map(friend => friend.name) : 'alone')`

Reply:
@kasionio 
For consistency, it is good that a function returns only certain type. So, make that in `true` case it also returns a string.
BTW, outer parenthses are not required in this example. It doesn't add to readability.

Reply:
@kasionio Make return statement consistent in types of returnable values and we are done

Reply:
Add `toString()` method in `true` case of ternary .

Thanks a lot! It was very helpful for me.